src/drc/api/serializers.py

```diff
@@ -397,6 +441,58 @@ def save(self, **kwargs):
         return self.instance
 
 
+class CompleteEnkelvoudigInformatieObjectSerializer(serializers.ModelSerializer):
+    """
+    Serializer for complete action of EnkelvoudigInformatieObject model
+    """
+    url = serializers.HyperlinkedIdentityField(
+        view_name='enkelvoudiginformatieobject-detail',
+        lookup_field='uuid'
+    )
+    inhoud = ViewFileFile(view_name='enkelvoudiginformatieobject-download', read_only=True)
+
+    class Meta:
+        model = EnkelvoudigInformatieObject
+        fields = ('url', 'inhoud', )
+
+    def validate(self, attrs):
+        valid_attrs = super().validate(attrs)
+
+        if not self.instance.canonical.lock:
+            raise serializers.ValidationError(
+                _("Unlocked document can't be marked as competed"),
+                code='unlocked'
+            )
+
+        if not self.instance.canonical.complete_upload:
+            raise serializers.ValidationError(
+                _("Upload of part files is not complete"),
+                code='incomplete-upload'
+            )
+
+        return valid_attrs
+
+    def save(self, **kwargs):
+        bestandsdelen = self.instance.canonical.bestandsdelen.order_by('index')
+        part_files = [p.inhoud.file for p in bestandsdelen]
+
+        # merge files
+        file_name = create_filename(self.instance.bestandsnaam)
+        file_dir = os.path.join(settings.PRIVATE_MEDIA_ROOT, datetime.now().strftime('uploads/%Y/%m/full/'))
```

Comment:
I think you can use the storage class of the field to prepare this for you, so you can more easily swap out the storage backend.

Take a look at how `django.models.fields.files.FileField.generate_filename` works.

I think something like:

```python
file_field = self.instance._meta.get_field("inhoud")
name = file_field.generate_filename(self.instance, file_name)
```

Should work

Reply:
Thanks! Added to the serializer